[UnifiedPDF] Implement HUD actions

hortont424 opened the pull request:
#### 86b022b127ea411def2339f0fec126ae7b9d36d6
<pre>
[UnifiedPDF] Implement HUD actions
<a href="https://bugs.webkit.org/show_bug.cgi?id=267910">https://bugs.webkit.org/show_bug.cgi?id=267910</a>
<a href="https://rdar.apple.com/118241818">rdar://118241818</a>

Reviewed by Richard Robinson.

Implement Open In... and Download in the HUD, and Open In... in the context menu.

* Source/WebKit/Shared/mac/PDFContextMenu.h:
* Source/WebKit/Shared/mac/PDFContextMenuItem.serialization.in:
* Source/WebKit/UIProcess/mac/WebPageProxyMac.mm:
(WebKit::WebPageProxy::showPDFContextMenu):
Rename openInPreviewIndex to openInPreviewTag since that&apos;s what it is in practice.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(WebKit::PDFPlugin::handleContextMenuEvent):
(WebKit::PDFPlugin::liveData const):
(WebKit::PDFPlugin::save): Deleted.
(WebKit::PDFPlugin::openWithPreview): Deleted.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm:
(WebKit::PDFPluginBase::createPDFDocument):
(WebKit::PDFPluginBase::originalData const):
(WebKit::PDFPluginBase::save):
(WebKit::PDFPluginBase::openWithPreview):
Hoist these to PDFPluginBase.

(WebKit::PDFPluginBase::rawData const): Deleted.
Rename rawData to originalData to properly convey what is special about it.

* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::liveData const):
Temporarily implement liveData() interms of originalData() until we have
annotation support implemented.

(WebKit::UnifiedPDFPlugin::createContextMenu const):
(WebKit::UnifiedPDFPlugin::save): Deleted.
(WebKit::UnifiedPDFPlugin::openWithPreview): Deleted.

Canonical link: <a href="https://commits.webkit.org/273346@main">https://commits.webkit.org/273346@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/78db03dc77a902809676963a8a41d8b7ca9fd47f

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/35155 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/14085 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/37277 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/37908 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/31732 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/16466 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/11153 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/30652 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/35698 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/11915 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/31351 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/10445 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/10503 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/39156 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/31978 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/31786 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/36500 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/10624 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/8551 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/34503 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/12398 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8048 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/11147 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/11460 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

aprotyas commented on Source/WebKit/UIProcess/mac/WebPageProxyMac.mm:
```diff
@@ -547,7 +547,7 @@ static inline bool expectsLegacyImplicitRubberBandControl()
     if (auto selectedMenuItem = [menuTarget selectedMenuItem]) {
         NSInteger tag = selectedMenuItem.tag;
-        if (contextMenu.openInPreviewIndex && *contextMenu.openInPreviewIndex == tag)
+        if (contextMenu.openInPreviewTag && *contextMenu.openInPreviewTag == tag)
```
I know you're just rewording things, but [`std::optional<T>::operator==(const optional<T>& opt, const U& value)`](https://en.cppreference.com/w/cpp/utility/optional/operator_cmp) does the null check for free

aprotyas reviewed (commented):
LGTM

hortont424 committed:
[UnifiedPDF] Implement HUD actions
https://bugs.webkit.org/show_bug.cgi?id=267910
rdar://118241818

Reviewed by Richard Robinson.

Implement Open In... and Download in the HUD, and Open In... in the context menu.

* Source/WebKit/Shared/mac/PDFContextMenu.h:
* Source/WebKit/Shared/mac/PDFContextMenuItem.serialization.in:
* Source/WebKit/UIProcess/mac/WebPageProxyMac.mm:
(WebKit::WebPageProxy::showPDFContextMenu):
Rename openInPreviewIndex to openInPreviewTag since that's what it is in practice.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPlugin.mm:
(WebKit::PDFPlugin::handleContextMenuEvent):
(WebKit::PDFPlugin::liveData const):
(WebKit::PDFPlugin::save): Deleted.
(WebKit::PDFPlugin::openWithPreview): Deleted.

* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.h:
* Source/WebKit/WebProcess/Plugins/PDF/PDFPluginBase.mm:
(WebKit::PDFPluginBase::createPDFDocument):
(WebKit::PDFPluginBase::originalData const):
(WebKit::PDFPluginBase::save):
(WebKit::PDFPluginBase::openWithPreview):
Hoist these to PDFPluginBase.

(WebKit::PDFPluginBase::rawData const): Deleted.
Rename rawData to originalData to properly convey what is special about it.

* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.h:
* Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm:
(WebKit::UnifiedPDFPlugin::liveData const):
Temporarily implement liveData() interms of originalData() until we have
annotation support implemented.

(WebKit::UnifiedPDFPlugin::createContextMenu const):
(WebKit::UnifiedPDFPlugin::save): Deleted.
(WebKit::UnifiedPDFPlugin::openWithPreview): Deleted.

Canonical link: https://commits.webkit.org/273346@main
